packages/js/components/src/experimental-select-control/select-control.tsx

```diff
@@ -121,45 +134,56 @@ function SelectControl< ItemType = DefaultItemType >( {
 		getItemProps,
 		selectItem,
 		selectedItem: comboboxSingleSelectedItem,
+		openMenu,
+		closeMenu,
 	} = useCombobox< ItemType | null >( {
 		initialSelectedItem: singleSelectedItem,
 		inputValue,
 		items: filteredItems,
+		selectedItem: multiple ? null : undefined,
 		itemToString: getItemLabel,
-		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-		// @ts-ignore
-		onStateChange: ( { inputValue: value, type, selectedItem } ) => {
+		onSelectedItemChange: ( { selectedItem } ) =>
+			selectedItem && onSelect( selectedItem ),
+		onInputValueChange: ( changes ) => {
+			if ( changes.inputValue !== undefined ) {
+				setInputValue( changes.inputValue );
+				if ( changes.isOpen ) {
```

Comment:
Just out of curiosity, we do not call `onInputChange` when the menu isn't open to avoid using this callback when resetting to the input value (to the selected value for single or blank when multiple)?

I don't have a strong opinion on this right now, but just wondering out loud whether or not we should always call this on change and let the consumer decide.  That might mean passing more information (`changes`) to the consumer though to let them decide.

Either way, I don't think it needs to hold up this PR and we can follow up on this.

Reply:
> Just out of curiosity, we do not call onInputChange when the menu isn't open to avoid using this callback when resetting to the input value (to the selected value for single or blank when multiple)?

correct.

> I don't have a strong opinion on this right now, but just wondering out loud whether or not we should always call this on change and let the consumer decide. That might mean passing more information (changes) to the consumer though to let them decide.

Yeah, that would work, passing back `changes` would make the most sense in that circumstance. My current interpretation was that our `onInputChange` callback only pertained to a user searching in the input field, which is why I stuck with that above logic. I don't mind changing this in the future though.

Reply:
>My current interpretation was that our onInputChange callback only pertained to a user searching in the input field, which is why I stuck with that above logic. 

Ya, that makes sense to me right now.  I'm not sure if we have a case that will need this callback fired under all circumstances, but we can leave it as is until that point comes.